### PR TITLE
feat: add onSave actions

### PR DESCRIPTION
**Pull request by victor-teles**
## Summary

This PR introduces some improvements to the IntelliJ plugin:

- Support for format on save
- Support for safe fixes on save
- Support for unsafe fixes on save
- Added native reformat with LSP
- Support for file filtering with glob pattern
- Format file with ctrl alt shift P/meta alt shift P (same as prettier plugin to make less impact for users that are migrating to Biome)
- Revamp configuration mode (disabled, automatic configuration, and manual configuration)

Sorry for the huge amount of changes 😅

## Automatic configuration mode
CleanShot 2023-12-19 at 19 06 44@2x

## Manual configuration mode
CleanShot 2023-12-19 at 19 07 33@2x

## Test Plan

Test saving actions
I'm planning to add UI testing later

### CODE REVIEW

File: src/main/kotlin/com/github/biomejs/intellijbiome/settings/BiomeConfigurable.kt [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+import java.util.regex.PatternSyntaxException
+import javax.swing.JCheckBox
+import javax.swing.JRadioButton
+import javax.swing.text.JTextComponent
+
+class BiomeConfigurable(internal val project: Project) :
+    BoundSearchableConfigurable(
+        BiomeBundle.message("biome.settings.name"),
+        BiomeBundle.message("biome.settings.name")
+    ) {
```
Comment (ematipico) [inline src/main/kotlin/com/github/biomejs/intellijbiome/settings/BiomeConfigurable.kt:37]: Is it correct to pass the same value twice?

File: src/main/kotlin/com/github/biomejs/intellijbiome/settings/BiomeConfigurable.kt [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+    }
+
+    private fun validateGlob(): ValidationInfoBuilder.(JBTextField) -> ValidationInfo? =
+        {
+            try {
+                FileSystems.getDefault().getPathMatcher("glob:" + it.text)
+                null
+            } catch (e: PatternSyntaxException) {
+                @NlsSafe val firstLine = e.localizedMessage?.lines()?.firstOrNull()
+                ValidationInfo(firstLine ?: BiomeBundle.message("biome.invalid.pattern"), it)
```
Comment (ematipico) [inline src/main/kotlin/com/github/biomejs/intellijbiome/settings/BiomeConfigurable.kt:204]: If the pattern is incorrect, does Biome stop working?

File: src/main/kotlin/com/github/biomejs/intellijbiome/settings/BiomeOnSaveInfoProvider.kt [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    override fun getActionOnSaveInfos(context: ActionOnSaveContext):
+        List<ActionOnSaveInfo> = listOf(
+        BiomeOnSaveFormatActionInfo(context),
+        BiomeOnSaveApplySafeFixesActionInfo(context),
+        BiomeOnSaveApplyUnsafeFixesActionInfo(context)
+    )
+
+    override fun getSearchableOptions(): Collection<String> {
+        return listOf(
+            BiomeBundle.message("biome.run.format.on.save.checkbox.on.actions.on.save.page"),
```
Comment (ematipico) [inline src/main/kotlin/com/github/biomejs/intellijbiome/settings/BiomeOnSaveInfoProvider.kt:18]: Out of curiosity, does the IDE give you autocomplete when you try to write a label?

File: src/main/kotlin/com/github/biomejs/intellijbiome/settings/BiomeConfigurable.kt (hunk 2 above)
Comment (victor-teles) [reply src/main/kotlin/com/github/biomejs/intellijbiome/settings/BiomeConfigurable.kt:204]: It's show a validation error

CleanShot 2023-12-20 at 16 43 15@2x

File: src/main/kotlin/com/github/biomejs/intellijbiome/settings/BiomeOnSaveInfoProvider.kt (hunk 3 above)
Comment (victor-teles) [reply src/main/kotlin/com/github/biomejs/intellijbiome/settings/BiomeOnSaveInfoProvider.kt:18]: Like this?
CleanShot 2023-12-20 at 16 53 03

Comment (ematipico) [reply src/main/kotlin/com/github/biomejs/intellijbiome/settings/BiomeOnSaveInfoProvider.kt:18]: Nice! Yeah that's exactly what I meant

**Review by ematipico — APPROVED** [review]

**Pull request merged**